src/spacel/provision/template/app_spot.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+logger = logging.getLogger('spacel.provision.template.app_spot')
+
+ASG_RESOURCES = ('Asg',
+                 'Lc')
+
+GDH_ASG_RESOURCE = ('SpScaleDown',
+                    'SpScaleDown',
+                    'AlarmScaleDown',
+                    'AlarmScaleUp')
+
+
+class AppSpotTemplateDecorator(object):
+    def spotify(self, app, region, template):
+        """
+        Replace ASG with SpotFleet. Literally spot-ify.
+        :return:
+        """
+        if app.spot is None:
+            return
+
+        resources = template['Resources']
+        self._add_spot_fleet(app, region, resources)
+        self._clean_up_asg(template)
+
+    @staticmethod
+    def _add_spot_fleet(app, region, resources):
+        spot_price = app.spot.get('price', '1.00')
+
+
+        # Extract parameters:
+        lc_properties = resources['Lc']['Properties']
+        user_data = lc_properties['UserData']
+        block_mapping = lc_properties['BlockDeviceMappings']
+        monitoring = {'Enabled': lc_properties['InstanceMonitoring']}
+        image_id = lc_properties['ImageId']
+        instance_profile = {
+            'Arn': {
+                'Fn::GetAtt': [lc_properties['IamInstanceProfile']['Ref'],
+                               'Arn']
+            }
+        }
+        security_groups = [
+            {'GroupId': {'Fn::GetAtt': [sg['Ref'], 'GroupId']}}
+            for sg in lc_properties['SecurityGroups']]
+
+        subnets = resources['Asg']['Properties']['VPCZoneIdentifier']
+
+        # Instance weights can be specified
+        weights = app.spot.get('weights')
+        if not weights:
+            weights = {app.instance_type: 1}
+
+        # If we're bidding on a single instance type, prefer AZ saturation
+        # else prefer lowest price.
+        if len(weights) == 1:
+            default_allocation_strat = 'diversified'
+        else:
+            default_allocation_strat = 'lowestPrice'
```

Comment:
this seems to be the wrong way around?

if we have more weights, we should go diversified (to make sure we balance between AZ's), if we have only one, just find the `lowestPrice`